exporter/common.go

```diff
@@ -91,42 +91,73 @@ func makeExcludeFilter(exclude []string) *primitive.E {
 	return &primitive.E{Key: "$and", Value: filterExpressions}
 }
 
-// makeDBsFilter creates a filter to list all databases or only the ones in the specified
-// namespaces list. Namespaces have the form of <db>.<collection> and the collection name
-// can have a dot. Example: db1.collection.one -> db: db1, collection: collection.one
-// db1, db2.col2, db3.col.one will produce [db1, db2, db3].
 func makeDBsFilter(filterInNamespaces []string) *primitive.E {
-	dbs := []string{}
-	if len(dbs) == 0 {
+	filterExpressions := []bson.D{}
+
+	nss := cleanupNamespaces(filterInNamespaces)
+	for _, namespace := range nss {
+		parts := strings.Split(namespace, ".")
+		filterExpressions = append(filterExpressions,
+			bson.D{{Key: "name", Value: bson.D{{Key: "$eq", Value: parts[0]}}}},
+		)
+	}
+
+	if len(filterExpressions) == 0 {
 		return nil
 	}
 
-	for _, namespace := range filterInNamespaces {
-		parts := strings.Split(namespace, ".")
-		dbs = append(dbs, parts[0])
+	return &primitive.E{Key: "$or", Value: filterExpressions}
+}
+
+func cleanupNamespaces(namespaces []string) []string {
```

Comment:
This method performs generic functionality of removing empty strings from a slice of strings.

Can we rename method to something generic (like **removeEmptyStrings**) and variable inside it along similar lines?